MSPI driver does not assign the 'op_mode' and 'duplex' in mspi_cfg

**Describe the bug**
in the description of the devicetree for the  [Multi-bit SPI Bus](https://docs.zephyrproject.org/latest/hardware/peripherals/mspi.html#device-tree) 
property  op-mode and duplex are supposed to be set  as follows

```
 			op-mode = "MSPI_CONTROLLER";
			duplex = "MSPI_HALF_DUPLEX";
```

The MSPI_CONFIG macro is capturing the mspi_cfg  with   (see drivers/mspi/mspi_emul.c)

```
		.op_mode = DT_ENUM_IDX_OR(n, op_mode, MSPI_OP_MODE_CONTROLLER),
		.duplex = DT_ENUM_IDX_OR(n, duplex, MSPI_HALF_DUPLEX),
```

which does not accept 
 			op-mode = "MSPI_CONTROLLER";
			duplex = "MSPI_HALF_DUPLEX";



**To Reproduce**
Steps to reproduce the behavior:
1. add overlay file for the mspi1 node with properties 'op-mode' and 'duplex' as above
2. west build
3. See error

**Expected behavior**
Use the appropriate DT macro to retrieve the mspi_config
```
mspi_config = 	
	{
		.channel_num = EMUL_MSPI_INST_ID, 		
		.op_mode = MSPI_OP_MODE_CONTROLLER,
		.duplex = MSPI_HALF_DUPLEX,
		etc.
	}
```

**Logs and console output**
```
devicetree_generated.h:8377:46: error: incompatible types when initializing type 'enum mspi_op_mode' using type 'char *'
...
in expansion of macro 'MSPI_BUS_NODE'
   37 |         .op_mode                  = DT_PROP_OR(MSPI_BUS_NODE, op_mode, MSPI_OP_MODE_CONTROLLER),


devicetree_generated.h:8393:45: error: incompatible types when initializing type 'enum mspi_duplex' using type 'char *'
...
 in expansion of macro 'MSPI_BUS_NODE'
   38 |         .duplex                   = DT_PROP_OR(MSPI_BUS_NODE, duplex, MSPI_HALF_DUPLEX),

```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows) ubuntu
 - Toolchain (e.g Zephyr SDK, ...) 0.16.8
 - Commit SHA or Version used 03959a20f7834ffe50a51df2b2e45ef87d1b610b

-->

--- a/drivers/mspi/mspi_emul.c
+++ b/drivers/mspi/mspi_emul.c
@@ -860,8 +860,8 @@ static struct emul_mspi_driver_api emul_mspi_driver_api = {
 #define MSPI_CONFIG(n)                                                                            \
 	{                                                                                         \
 		.channel_num           = EMUL_MSPI_INST_ID,                                       \
-		.op_mode               = DT_ENUM_IDX_OR(n, op_mode, MSPI_OP_MODE_CONTROLLER),     \
-		.duplex                = DT_ENUM_IDX_OR(n, duplex, MSPI_HALF_DUPLEX),             \
+		.op_mode               = DT_INST_ENUM_IDX_OR(n, op_mode, MSPI_OP_MODE_CONTROLLER),\
+		.duplex                = DT_INST_ENUM_IDX_OR(n, duplex, MSPI_HALF_DUPLEX),        \
 		.max_freq              = DT_INST_PROP(n, clock_frequency),                        \
 		.dqs_support           = DT_INST_PROP_OR(n, dqs_support, false),                  \
 		.sw_multi_periph       = DT_INST_PROP(n, software_multiperipheral),               \

--- a/tests/drivers/mspi/api/src/main.c
+++ b/tests/drivers/mspi/api/src/main.c
@@ -31,8 +31,8 @@ static struct gpio_dt_spec ce_gpios[] = MSPI_CE_GPIOS_DT_SPEC_GET(MSPI_BUS_NODE)
 #if TEST_MSPI_REINIT
 struct mspi_cfg hardware_cfg = {
 	.channel_num              = 0,
-	.op_mode                  = DT_PROP_OR(MSPI_BUS_NODE, op_mode, MSPI_OP_MODE_CONTROLLER),
-	.duplex                   = DT_PROP_OR(MSPI_BUS_NODE, duplex, MSPI_HALF_DUPLEX),
+	.op_mode                  = DT_ENUM_IDX_OR(MSPI_BUS_NODE, op_mode, MSPI_OP_MODE_CONTROLLER),
+	.duplex                   = DT_ENUM_IDX_OR(MSPI_BUS_NODE, duplex, MSPI_HALF_DUPLEX),
 	.dqs_support              = DT_PROP_OR(MSPI_BUS_NODE, dqs_support, false),
 	.ce_group                 = ce_gpios,
 	.num_ce_gpios             = ARRAY_SIZE(ce_gpios),

